quantestpy/circuit.py

```diff
@@ -385,41 +344,13 @@ def assert_equal(
     whole_gates_a = test_circuit_a._get_whole_gates()
     whole_gates_b = test_circuit_b._get_whole_gates()
 
-    if matrix_norm_type is None:
-        # assert equal exact or equal up to a global phase
-        operator.assert_equal(
-            whole_gates_a,
-            whole_gates_b,
-            rtol,
-            atol,
-            up_to_global_phase,
-            msg)
-
-    else:
-        # assert check matrix norm as a distance
-        matrix_norm_a_minus_b = _get_matrix_norm(
-            whole_gates_a,
-            whole_gates_b,
-            matrix_norm_type,
-            up_to_global_phase
-        )
-
-        if rtol != 0.:
-            matrix_norm_b = _get_matrix_norm(
-                whole_gates_b,
-                np.zeros_like(whole_gates_b),
-                matrix_norm_type,
-                False
-            )
-
-        else:
-            matrix_norm_b = 0.
-
-        if matrix_norm_a_minus_b >= atol + rtol * matrix_norm_b:
-
-            error_msg = "matrix norm ||A-B|| " \
-                + format(matrix_norm_a_minus_b, ".15g") \
-                + " is larger than (atol + rtol*||B||) " \
-                + format(atol + rtol * matrix_norm_b, ".15g") + "."
-            msg = ut_test_case._formatMessage(msg, error_msg)
-            raise QuantestPyAssertionError(msg)
+    # call operator.assert_equal
+    operator.assert_equal(
+        whole_gates_a,
+        whole_gates_b,
+        rtol,
+        atol,
+        up_to_global_phase,
+        matrix_norm_type,
+        msg
+    )
```

Comment:
Now `operator.assert_equal()` is called both in `circuit.assert_equal()` and in `circuit.assert_equal_to_operator()`, which enables us to maintain the package easily.